orangecontrib/infrared/widgets/owfft.py

```diff
@@ -0,0 +1,594 @@
+import sys
+import numpy as np
+
+from PyQt4.QtGui import QGridLayout
+
+import Orange.data
+from Orange.widgets.widget import OWWidget
+from Orange.widgets import gui, settings
+
+def build_spec_table(wavenumbers, intensities):
+    """
+    Converts numpy arrays of wavenumber and intensity into an
+    Orange.data.Table spectra object.
+
+    Args:
+        wavenumbers (np.array): 1D array of wavenumbers
+        intensities (np.array): 2D array of (multi-spectra) intensities
+
+    Returns:
+        table: Orange.data.Table object, spectra format
+    """
+
+    # Add dimension to 1D array if necessary
+    if intensities.ndim == 1:
+        intensities = intensities[None,:]
+
+    # Convert the wavenumbers array into a list of ContinousVariables
+    wn_vars = [Orange.data.ContinuousVariable.make(repr(wavenumbers[i]))
+                for i in range(wavenumbers.shape[0])]
+
+    # Build an Orange.data.Domain object with wn_vars as
+    # independant variables (or "attributes" as Orange calls them)
+    domain = Orange.data.Domain(wn_vars)
+
+    # Finally, build the table using the damain and intensity arrays:
+    table = Orange.data.Table.from_numpy(domain, intensities)
+    return table
+
+class OWFFT(OWWidget):
+    # Widget's name as displayed in the canvas
+    name = "FFT"
```

Comment:
I would call it something else than FFT. Maybe IFG->SP would be a good one, since it has many options/operations unrelated to FFT and the functionality is that...